stdlib/src/base64/_b64encode.mojo

```diff
@@ -0,0 +1,408 @@
+# ===----------------------------------------------------------------------=== #
+# Copyright (c) 2024, Modular Inc. All rights reserved.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions:
+# https://llvm.org/LICENSE.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===----------------------------------------------------------------------=== #
+
+"""
+We make use of the following papers for the implementation, note that there
+are some small differences.
+Wojciech Muła, Daniel Lemire, Base64 encoding and decoding at almost the
+speed of a memory copy, Software: Practice and Experience 50 (2), 2020.
+https://arxiv.org/abs/1910.05109
+
+Wojciech Muła, Daniel Lemire, Faster Base64 Encoding and Decoding using AVX2
+Instructions, ACM Transactions on the Web 12 (3), 2018.
+https://arxiv.org/abs/1704.00605
+"""
+
+from collections import InlineArray
+from memory import memcpy, bitcast, UnsafePointer
+from memory.maybe_uninitialized import UnsafeMaybeUninitialized
+
+
+fn _subtract_with_saturation[
```

Comment:
There is a LLVM intrinsics powered `_sub_with_saturation` (#3654) in `_utf8_validation.mojo`.

Reply:
Good call. I put this function in the `simd.mojo` file as a private function to avoid duplication.